python/triqs/gf/gf.py

```diff
@@ -302,10 +302,37 @@ def __getitem__(self, key):
         if key == self._full_slice:
             return self
 
-        # Only one argument. Must be a mesh point
         if not isinstance(key, tuple):
-            assert isinstance(key, (MeshPoint, Idx))
-            return self.data[key.linear_index if isinstance(key, MeshPoint) else self._mesh.index_to_linear(key.idx)]
+            if isinstance(key, (MeshPoint, Idx)):
+                return self.data[key.linear_index if isinstance(key, MeshPoint) else self._mesh.index_to_linear(key.idx)]
+            else:
+                assert self.target_rank == 1, "wrong number of arguments. Expected %s, got %s"%(self.target_rank, 1)
+                # Assume empty indices (scalar_valued)
+                ind = GfIndices([])
+
+                # String access: transform the key into a list integers
+                if isinstance(key, str):
+                    assert self._indices, "Got string indices, but I have no indices to convert them !"
+                    key_lst = [self._indices.convert_index(key,1)] # convert returns a slice of len 1
```

Comment:
I am not sure if this is necessary, as I don't know any possibility of constructing a rank1 Gf with string indices...